File: tests/conformance/fixtures/python/11-full-options.py
from sykli import Pipeline

p = Pipeline()

p.task("test").run("npm test") \
    .container("node:20") \
    .workdir("/app") \
    .env("NODE_ENV", "test") \
    .env("CI", "true") \
    .retry(3) \
    .timeout(300) \
    .secrets("NPM_TOKEN", "GH_TOKEN") \
    .when("branch:main")

p.emit()

File: tests/conformance/fixtures/rust/11-full-options.rs
use sykli::Pipeline;

fn main() {
    let mut p = Pipeline::new();
    p.task("test").run("npm test")
        .container("node:20")
        .workdir("/app")
        .env("CI", "true")
        .env("NODE_ENV", "test")
        .retry(3)
        .timeout(300)
        .secrets(&["NPM_TOKEN", "GH_TOKEN"])
        .when("branch:main");
    p.emit();
}
